src/NzbDrone.Common/Disk/DiskTransferService.cs

```diff
@@ -0,0 +1,267 @@
+        }
+
+        private Boolean TryCopyFile(String sourcePath, String targetPath)
+        {
+            var originalSize = _diskProvider.GetFileSize(sourcePath);
+
+            for (var i = 0; i <= RetryCount; i++)
+            {
+                _diskProvider.CopySingleFile(sourcePath, targetPath);
+
+                var targetSize = _diskProvider.GetFileSize(targetPath);
+
+                if (targetSize == originalSize)
+                {
+                    return true;
+                }
+
+                _diskProvider.DeleteFile(targetPath);
+
+                if (i == RetryCount)
+                {
+                    _logger.Error("Failed to completely transfer [{0}] to [{1}], aborting.", sourcePath, targetPath, i + 1, RetryCount);
+                }
+                else if (i == 0)
+                {
+                    _logger.Warn("Failed to completely transfer [{0}] to [{1}], retrying [{2}/{3}].", sourcePath, targetPath, i + 1, RetryCount);
+                }
+                else
+                {
+                    _logger.Debug("Failed to completely transfer [{0}] to [{1}], retrying [{2}/{3}].", sourcePath, targetPath, i + 1, RetryCount);
+                }
+            }
+
+            return false;
+        }
+
+        private Boolean TryMoveFile(String sourcePath, String targetPath)
+        {
+            var originalSize = _diskProvider.GetFileSize(sourcePath);
+
+            var backupPath = sourcePath + ".movebackup";
+
+            if (_diskProvider.FileExists(backupPath))
+            {
+                _logger.Trace("Removing old backup.");
+                _diskProvider.DeleteFile(backupPath);
+            }
+
+            try
+            {
+                _logger.Trace("Attempting to move hardlinked backup.");
+                if (_diskProvider.TryCreateHardLink(sourcePath, backupPath))
+                {
+                    _diskProvider.MoveSingleFile(backupPath, targetPath);
+
+                    var targetSize = _diskProvider.GetFileSize(targetPath);
+
+                    if (targetSize == originalSize)
+                    {
+                        _logger.Trace("Hardlink move succeeded, deleting source.");
+                        _diskProvider.DeleteFile(sourcePath);
+                        return true;
+                    }
+                }
+            }
+            finally
+            {
+                if (_diskProvider.FileExists(backupPath))
+                {
+                    _diskProvider.DeleteFile(backupPath);
+                }
+            }
+
+            _logger.Trace("Hardlink move failed, reverting to copy.");
+            if (TryCopyFile(sourcePath, targetPath))
+            {
+                _logger.Trace("Copy succeeded, deleting source.");
+                _diskProvider.DeleteFile(sourcePath);
+                return true;
+            }
+
+            _logger.Trace("Copy failed.");
```

Comment:
Are you sure? Most, if not all, paths are already logged by the caller of this method.